### PR TITLE
[FR] Simplify / optimize setting loading 

### LINKED ISSUE #3982: [FR] Settings allocator
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### Problem statement

Currently, it is not always clear which config plane set a specific setting.

### Suggested solution

Exposing that info with the API would be great.
My implementation idea would be to add a dict that is filled on each config call and that gets exposed with API calls.

### Describe alternatives you've considered

You can search by hand rn.

### Examples of other systems

_No response_

### Do you want to develop this?

- [x] I want to develop this.

### PR DESCRIPTION
**Pull request by matmair**
This PR adds a global config-file cache. Environment variables will still be read live to enable CI manipulation etc.

Fixes https://github.com/inventree/InvenTree/issues/4149
Fixes #3982

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/4152"></a>

### CODE REVIEW

**Comment by matmair** [comment]
@SchrodingersGat ready for review

File: InvenTree/InvenTree/config.py [hunk 1]
```diff
@@ -56,8 +59,17 @@ def get_config_file(create=True) -> Path:
...
+    """Load configuration data from the config file.
+
+    Arguments:
+        set_cache(bool): If True, the configuration data will be cached for future use after load.
+    """
+
+    # use cache if populated
+    # skip cache if cache should be set
+    if CONIFG_DATA is not None and not set_cache:
+        return CONIFG_DATA
```
Comment (SchrodingersGat) [inline InvenTree/InvenTree/config.py:72]: spelling :)

**Review by SchrodingersGat — CHANGES_REQUESTED** [review]

File: InvenTree/InvenTree/config.py [hunk 2]
```diff
@@ -8,6 +9,8 @@
 from pathlib import Path
 
 logger = logging.getLogger('inventree')
+CONIFG_DATA = None
```
[the remark this answers is not in the document]
Comment (matmair) [reply InvenTree/InvenTree/config.py:12]: Fixed :+1

**Review by SchrodingersGat — APPROVED** [review]

**Comment by SchrodingersGat** [comment]
Great work thanks @matmair

**Pull request merged**